pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -327,14 +325,12 @@ private void internalStoreDelete(OpDelete op) {
 
         try {
             zkc.delete(op.getPath(), expectedVersion, (rc, path1, ctx) -> {
-                execute(() -> {
```

Comment:
Use the execute is unnecessary, the executor is only used to execute flush, or add the request to the queue in zk callback.